crates/bevy_ecs/src/archetype.rs

```diff
@@ -766,7 +766,7 @@ pub struct Archetypes {
     /// find the archetype id by the archetype's components
     by_components: HashMap<ArchetypeComponents, ArchetypeId>,
     /// find all the archetypes that contain a component
-    by_component: ComponentIndex,
+    pub(crate) by_component: ComponentIndex,
```

Comment:
Should this maybe be a new method on `Archetypes`? Like `fn containing_component(&self, component: ...) -> ...`
I don't like leaking details to the rest of the crate as it will make internal refactorings harder. The principle invoked here is often called "Talk to friends, not to strangers"

Reply:
Ah, I see. No, it's fine then :)